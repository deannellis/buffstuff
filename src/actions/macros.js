import database from '../firebase/firebase';

export const setMacros = (current = {}, target = {}) => ({
	type: 'SET_MACROS',
	target,
	current,
});

export const startFetchMacros = () => {
	return (dispatch) => {
		return database
			.ref('macros')
			.once('value')
			.then((snapshot) => {
				if (snapshot.val() !== null) {
					const { target, current } = snapshot.val();
					dispatch(setMacros(current, target));
				}
			});
	};
};

export const setTargetMacros = ({ protein, carbs, fat }) => ({
	type: 'SET_TARGET_MACROS',
	target: {
		protein,
		carbs,
		fat,
	},
});

export const startSetTargetMacros = ({
	protein = 0,
	carbs = 0,
	fat = 0,
} = {}) => {
	return (dispatch) => {
		return database
			.ref('macros')
			.set({
				target: { protein, carbs, fat },
				current: { protein: 0, carbs: 0, fat: 0 },
			})
			.then(() => {
				dispatch(setTargetMacros({ protein, carbs, fat }));
			});
	};
};

export const updateMacro = ({ macro, amount }) => ({
	type: 'UPDATE_MACRO',
	update: {
		macro,
		amount,
	},
});

export const startUpdateMacro = ({ macro = '', amount = 0 } = {}, current) => {
	return (dispatch) => {
		const newAmount = current[macro] + amount;
		return database
			.ref(`macros/current/${macro}`)
			.set(newAmount)
			.then(() => {
				dispatch(updateMacro({ macro, amount }));
			});
	};
};

export const setCurrentDate = ({ currentDate = 0 } = {}) => ({
	type: 'SET_CURRENT_DATE',
	currentDate,
});

export const resetCurrent = () => ({ type: 'RESET_CURRENT' });

export const startResetCurrent = () => {
	return (dispatch) => {
		return database
			.ref('macros/current')
			.set({ protein: 0, carbs: 0, fat: 0 })
			.then(() => {
				dispatch(resetCurrent());
			});
	};
};

// export const syncMacros = () => ({ type: 'SYNC_MACROS' });

// export const startSyncMacros = () => {
// 	return (dispatch) => {
// 		console.log('called!!!');
// 		return database
// 			.ref('macros')
// 			.set(macrosReducerDefaultState)
// 			.then(() => {
// 				dispatch(syncMacros());
// 			});
// 	};
// };
