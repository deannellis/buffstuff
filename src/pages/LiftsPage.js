import React, { Component } from 'react';
import { withRouter } from "react-router-dom";
import { connect } from 'react-redux';
import SideNav from '../components/SideNav';
import Tabs from '../components/Tabs';
import LineGraph from '../components/LineGraph';

class LiftsPage extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            activeTab: 0
        }
    }

    getData = () => {
        const { workouts } = this.props;
        let filteredWorkouts = [];
        workouts.map(workout => {
            let workoutWithDate = {}
            if(workout.hasOwnProperty('squat')) {
                workoutWithDate = {
                    ...workout['squat'],
                    date: workout.created
                }
                filteredWorkouts.push(workoutWithDate)
            }
        })
        // console.log('fw', filteredWorkouts);
        return filteredWorkouts
    }

    render() { 
        return (
            <div className="page--with-side-nav">
                <SideNav path={this.props.match.path} />
                <div className="lifts-page">
                    <Tabs 
                        activeIndex={this.state.activeTab} 
                        handleSelect={i => this.setState({ activeTab: i })}
                        labels={['Bench Press', 'Deadlift', 'Overhead Press', 'Row', 'Squat']}
                    >
                        <LineGraph data={this.getData()}/>

                        {/* {this.state.activeTab === 0 && 
                            <p>1</p>
                        }
                        {this.state.activeTab === 1 && 
                            <p>2</p>
                        }
                        {this.state.activeTab === 2 && 
                            <p>3</p>
                        }
                        {this.state.activeTab === 3 && 
                            <p>4</p>
                        }
                        {this.state.activeTab === 4 && 
                            <p>5</p>
                        } */}
                    </Tabs>
                </div>
            </div>
        );
    }
}
 
const mapStateToProps = state => {
    return {
        workouts: state.workouts,
    };
}
 
export default withRouter(connect(mapStateToProps)(LiftsPage));