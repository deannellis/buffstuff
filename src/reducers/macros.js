const macrosReducerDefaultState = {
	target: {
		protein: null,
		carbs: null,
		fat: null
	},
	current: {
		protein: null,
		carbs: null,
		fat: null
	},
	dateObject: null
};

const macrosReducer = (state = macrosReducerDefaultState, action) => {
	let setTargetUpdates = {};
	const { update } = action;
	const updateMacroUpdates = { ...state };
	const { currentDate } = action;
	const setCurrentDateUpdates = { ...state };
	const resetCurrentUpdates = {
		target: { ...state.target },
		current: { ...macrosReducerDefaultState.current },
		dateObject: null
	};
	switch (action.type) {
		case 'SET_TARGET_MACROS':
			setTargetUpdates = {
				target: { ...action.target },
				current: { ...state.current },
				dateObject: { ...state.dateObject }
			};
			return setTargetUpdates;
		case 'UPDATE_MACRO':
			if (updateMacroUpdates.current[update.macro] !== null) {
				updateMacroUpdates.current[update.macro] += update.amount;
			} else {
				updateMacroUpdates.current[update.macro] = update.amount;
			}
			return updateMacroUpdates;
		case 'SET_CURRENT_DATE':
			setCurrentDateUpdates.dateObject = currentDate;
			return setCurrentDateUpdates;
		case 'RESET_CURRENT':
			return resetCurrentUpdates;
		default:
			return state;
	}
};

export default macrosReducer;
