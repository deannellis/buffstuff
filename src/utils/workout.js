import moment from 'moment';

export const getWorkouts = ({ a = 0, b = 0 }) => {
    let workouts = []
    if(a == 0) {
        workouts.push('overhead', 'chinup')
    } else {
        workouts.push('bench', 'row')
    }
    if(b == 0 || b == 2) {
        workouts.push('squat');
    } else {
        workouts.push('deadlift');
    }
    return workouts;
}

export const getDisplayName = name => {
    switch(name) {
        case 'bench':
            return 'Bench Press';
        case 'row':
            return 'Row';
        case 'squat':
            return 'Squat';
        case 'deadlift':
            return 'Deadlift';
        case 'overhead':
            return 'Overhead Press';
        case 'chinup':
            return 'Chin-Ups';
        default:
            return name;
    }
};

export const getWarmupWeights = workWeight => {
    const warmup25 = workWeight * 0.25 > 44 ? workWeight * 0.25 : 44;
    const warmup50 = workWeight * 0.50 > 44 ? workWeight * 0.50 : 44;
    const warmup75 = workWeight * 0.75 > 44 ? workWeight * 0.75 : 44;
    return [ warmup25, warmup50, warmup75 ];
}

export const getWeightDistribution = weight => {
    if(weight === 44) {
        return [ 0, 44, 0]
    }
    const oneSide = (weight - 44) / 2;
    return [ oneSide, 44, oneSide ];
}

export const getMonthWorkouts = workouts => {
    let dates = {};
    workouts.forEach(workout => {
        let month = moment(workout.created).format("MMMM");
        let day = parseInt(moment(workout.created).format("D"));
        if(dates.hasOwnProperty(month)) {
            dates[month].push(day);
        } else {
            dates[month] = [ day ];
        }
    });
    // console.log(dates);
    return dates;
}

export const getWorkoutDays = (start, daysInMonth) => {
    let workoutDays = [];
    let checkEven = false;
    let weekday = start + 1;
    for(let d = 1; d <= daysInMonth; d++) {
        if(checkEven) {
            if(d%2 == 0) workoutDays.push(d)
        } else {
            if(d%2 !== 0) workoutDays.push(d)
        }
        if(weekday == 7){
            weekday = 0;
            checkEven = !checkEven;
        };
        weekday ++;
    }
    return workoutDays;
};
