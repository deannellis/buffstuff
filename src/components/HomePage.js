import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from "react-router-dom";
import { startWorkout } from '../actions/inProgressWorkout';
import Tabs from './Tabs';
import Dashboard from './Dashboard';
import WorkoutsTab from './WorkoutsTab';

class HomePage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            activeTab: 1
        }
    }

    onStartWorkout = () => {
        this.props.dispatch(startWorkout(this.props.liftVariant));
        this.props.history.push('/workout');
    }

    componentDidMount() {
        if(Object.entries(this.props.masterWeights).length === 0 && this.props.masterWeights.constructor === Object) {
            this.props.history.push('/onboarding');
        }
    }

    render() { 
        return (
            <>
                <Tabs 
                    activeIndex={this.state.activeTab} 
                    handleSelect={i => this.setState({ activeTab: i })}
                    labels={['Dashboard', 'Workouts']}
                >
                    {this.state.activeTab === 0 && 
                        <Dashboard
                            liftVariant={this.props.liftVariant}
                            masterWeights={this.props.masterWeights} 
                            workouts={this.props.workouts}
                            onStartWorkout={this.onStartWorkout}
                        />
                    }
                    {this.state.activeTab === 1 && 
                        <WorkoutsTab 
                            workouts={this.props.workouts}
                        />
                    }
                </Tabs>
            </>
        );
    }
}

const mapStateToProps = state => {
    return {
        masterWeights: state.masterWeights,
        liftVariant: state.liftVariant,
        workouts: state.workouts,
    };
}
 
export default withRouter(connect(mapStateToProps)(HomePage));