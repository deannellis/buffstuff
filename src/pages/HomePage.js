import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, Link} from "react-router-dom";
import moment from 'moment';

import { startWorkout } from '../actions/inProgressWorkout';
import { setTargetMacros, updateMacro, setCurrentDate, resetCurrent } from '../actions/macros';
import Dashboard from '../components/Dashboard';
import Button from '../components/Button';
import SideNav from '../components/SideNav';

class HomePage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            today: moment()
        }
    }

    onStartWorkout = () => {
        this.props.dispatch(startWorkout(this.props.liftVariant));
        this.props.history.push('/workout');
    }

    onSetMacros = macros => { this.props.dispatch(setTargetMacros(macros)) }

    onUpdateMacro = update => {
        this.props.dispatch(updateMacro(update))
    }

    componentDidMount() {
        const { dateObject } = this.props.macros;
        const storedDate = dateObject._d
        const { today } = this.state;
        if(dateObject === null || dateObject === 0) {
            console.log('SETTING DATE DUE TO NULL OR 0 VALUE');
            this.props.dispatch(setCurrentDate({ currentDate: today }));
        }else if(!(moment().isSame(storedDate, 'day'))) {
            console.log('RESETING CURRENT MACROS BASED ON OUTDATED DATE');
            this.props.dispatch(resetCurrent());
            this.props.dispatch(setCurrentDate({ currentDate: today }));
        }
    }

    render() { 
        if(Object.entries(this.props.masterWeights).length === 0 && this.props.masterWeights.constructor === Object) {
            return (
                <>
                    <h2>Welcome to flext!</h2>
                    <p>Let's begin by entering your starting weights</p>
                    <Link to="/onboarding">
                        <Button variant="primary">
                            Enter Weights
                        </Button>
                    </Link>
                </>
            );
        }
        return (
            <div className="page--with-side-nav">
                <SideNav path={this.props.match.path} />
                <Dashboard
                    liftVariant={this.props.liftVariant}
                    masterWeights={this.props.masterWeights} 
                    workouts={this.props.workouts}
                    onStartWorkout={this.onStartWorkout}
                    macros={this.props.macros}
                    onSetMacros={this.onSetMacros}
                    onUpdateMacro={this.onUpdateMacro}
                />
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        masterWeights: state.masterWeights,
        liftVariant: state.liftVariant,
        workouts: state.workouts,
        macros: state.macros,
    };
}
 
export default withRouter(connect(mapStateToProps)(HomePage));