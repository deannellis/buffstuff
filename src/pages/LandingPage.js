import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { startLogin } from '../actions/auth';
import Button from '../components/Button';
import AppFooter from '../components/AppFooter';

const LoginPage = ({ startLogin }) => {
	return (
		<div className="landing-page">
			<div className="landing-page__hero">
				<div>
					<h1>flext takes the guess work out of weight lifting</h1>
					<Button clickHandler={startLogin()}>
						<img
							src="./images/64px-Google__G__Logo.webp"
							alt="Google mark"
							height="32px"
						/>
						Log in with Google
					</Button>
				</div>
				<div className="landing-page__hero-image">
					<img
						src="./images/landingIllustration.png"
						alt="illustration of weightlifter"
					/>
				</div>
			</div>
			<div className="landing-page__section-2">
				<div>
					<img
						src="./images/workoutComponents.png"
						alt="Components from within the application"
					/>
				</div>
				<div>
					<h1>
						flext knows what your next workout looks like so you don&apos;t have
						to
					</h1>
				</div>
			</div>
			<div className="landing-page__section-3">
				<div>
					<h1>See your gains increase as you progress</h1>
				</div>
				<div>
					<img
						src="./images/weightGraph.png"
						alt="Components from within the application"
					/>
				</div>
			</div>
			<div className="landing-page__section-4">
				<div>
					<img
						src="./images/macroComponent.png"
						alt="Components from within the application"
					/>
				</div>
				<div>
					<h1>Track your macros to optimize your gains</h1>
				</div>
			</div>
			<AppFooter />
		</div>
	);
};

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
