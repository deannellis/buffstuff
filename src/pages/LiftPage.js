import React, { Component } from 'react';
import { withRouter } from "react-router";
import { connect } from 'react-redux';
import { updateWorkout } from '../actions/inProgressWorkout';
import { getDisplayName, getWarmupWeights, getWeightDistribution } from '../utils/workout';
import Button from '../components/Button';

class LiftPage extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }

    completeLift = (result) => {
        const liftResult = {updates: {[this.props.match.params.id]: result}}
        console.log(liftResult);
        this.props.dispatch(updateWorkout(liftResult));
        this.props.history.push('/workout');
    }

    render() { 
        let { id } = this.props.match.params;
        const workWeight = this.props.masterWeights[id];
        const workReps = ['5 reps', '5 reps', 'AMRAP*'];
        const { negatives, ups, weight } = this.props.masterWeights.chinup;
        return (
            <div className="lift-page">
                <div className="lift-page__card">
                    <h2>
                        {getDisplayName(id)}
                    </h2>
                    {id === 'chinup' ? (
                        <div>
                            <h3>Complete</h3>
                            <p className="lift-page__chinup-title">3 sets of</p>
                            <div className="lift-page__chinups">
                                {negatives === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{negatives}</p>
                                        <span>negatives</span>
                                    </div>
                                )}
                                {ups === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{ups}</p>
                                        <span>chin-ups</span>
                                    </div>
                                )}
                                {weight === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{weight}</p>
                                        <span>lbs</span>
                                    </div>
                                )}
                            </div>
                            <h3>Enter your results</h3>
                            <p>Were you able to complete all sets and reps?</p>
                            <Button variant="primary" clickHandler={() => {this.completeLift(1)}}>Yes</Button>
                            <Button clickHandler={() => {this.completeLift(0)}}>No</Button>
                        </div>
                    ) : (
                        <>
                            <div className="lift-page__tables">
                                <div>
                                    <h3>Warm-Up</h3>
                                    {getWarmupWeights(workWeight).map((weight, i) => (
                                        <div key={i} className="lift-page__sets">
                                            <div className="lift-page__set">
                                                <span>Set {i + 1}</span>
                                                <p>5 reps @{weight}lbs</p>
                                            </div>
                                            <div className="lift-page__distributed">
                                                {getWeightDistribution(weight).map((distWeight, i) => (
                                                    <div key={i} className="lift-page__distributed-item">
                                                        <span>
                                                            {i === 0 && 'LEFT'}
                                                            {i === 1 && 'BAR'}
                                                            {i === 2 && 'RIGHT'}
                                                        </span>
                                                        <p>{distWeight}</p>
                                                    </div>
                                                ))}
                                            </div>
                                        </div>
                                    ))}
                                </div>
                                <div>
                                    <h3>Work</h3>
                                    {workReps.map((rep, i) => (
                                        <div key={i} className="lift-page__sets">
                                                <div className="lift-page__set">
                                                    <span>Set {i + 1}</span>
                                                    <p>{rep} @{workWeight}lbs</p>
                                                </div>
                                                <div className="lift-page__distributed">
                                                    {getWeightDistribution(workWeight).map((distWeight, i) => (
                                                        <div key={i} className="lift-page__distributed-item">
                                                            <span>
                                                                {i === 0 && 'LEFT'}
                                                                {i === 1 && 'BAR'}
                                                                {i === 2 && 'RIGHT'}
                                                            </span>
                                                            <p>{distWeight}</p>
                                                        </div>
                                                    ))}
                                                </div>
                                        </div>
                                    ))}
                                </div>
                            </div>
                            <p>*as many reps as possible </p>
                            <h3 className="lift-page__results-title">Enter your results</h3>
                            <p>On your last set how many reps did you complete?</p>
                            <Button clickHandler={() => {this.completeLift(0)}}>Less than 5</Button>
                            <Button clickHandler={() => {this.completeLift(1)}}>Greater than 5</Button>
                            <Button clickHandler={() => {this.completeLift(2)}}>Greater than 10</Button>
                        </>
                    )}
                    
                </div>
            </div>
        );
    }
}
 
const mapStateToProps = state => {
    return {
        masterWeights: state.masterWeights,
    };
}
 
export default withRouter(connect(mapStateToProps)(LiftPage));