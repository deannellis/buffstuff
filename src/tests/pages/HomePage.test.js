import React from 'react';
import { shallow } from 'enzyme';

import { HomePage } from '../../pages/HomePage';
import { weights, workouts } from '../fixtures/workout';
import { macros } from '../fixtures/macros';

test('should render Home Page without props', () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper).toMatchSnapshot();
});

test('should render Home Page with props', () => {
    const wrapper = shallow(<HomePage 
        masterWeights={weights}  
        liftVariant={{ a:1, b:1 }}
        workouts={workouts}
        macros={macros}
    />);
    expect(wrapper).toMatchSnapshot();
});

test('should render Home Page with welcome message', () => {
    const wrapper = shallow(<HomePage 
        masterWeights={{}}  
        liftVariant={{ a:0, b:0 }}
        workouts={[]}
        macros={macros}
    />);
    expect(wrapper).toMatchSnapshot();
});
