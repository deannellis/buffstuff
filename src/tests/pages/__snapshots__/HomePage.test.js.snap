// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home Page with props 1`] = `
<div
  className="page--with-side-nav"
>
  <SideNav
    path=""
  />
  <div
    className="side-nav__page-content"
  >
    <div
      className="side-nav__page-scrim side-nav__page-scrim--hidden"
    />
    <div
      className="dashboard__container side-nav__page-content"
    >
      <NextWorkout
        liftVariant={
          Object {
            "a": 1,
            "b": 1,
          }
        }
        masterWeights={
          Object {
            "bench": 85,
            "chinup": Object {
              "negatives": 2,
              "ups": 3,
              "weight": 0,
            },
            "deadlift": 106,
            "overhead": 112,
            "row": 115,
            "squat": 156,
          }
        }
        onStartWorkout={[Function]}
      />
      <div
        className="card"
      >
        <Calendar
          onClickWorkoutDate={[Function]}
          workouts={
            Array [
              Object {
                "chinup": Object {
                  "result": 1,
                  "weight": Object {
                    "negatives": 1,
                    "ups": 4,
                    "weight": 0,
                  },
                },
                "created": "2020-02-08T18:28:12.461Z",
                "id": "1d8ad17a-c62a-4182-8ddb-0df62af33c65",
                "overhead": Object {
                  "result": 2,
                  "weight": 112,
                },
                "squat": Object {
                  "result": 2,
                  "weight": 156.6,
                },
              },
              Object {
                "bench": Object {
                  "result": 2,
                  "weight": 85,
                },
                "created": "2020-02-06T18:41:49.124Z",
                "deadlift": Object {
                  "result": 1,
                  "weight": 106,
                },
                "id": "7a07e348-e0e8-48c8-a60a-3d9fb9929ab5",
                "row": Object {
                  "result": 2,
                  "weight": 115.5,
                },
              },
              Object {
                "chinup": Object {
                  "result": 0,
                  "weight": Object {
                    "negatives": 2,
                    "ups": 3,
                    "weight": 0,
                  },
                },
                "created": "2020-02-03T21:01:59.832Z",
                "id": "de5f7586-8023-4b3a-bfcf-0ff72fe2c4d8",
                "overhead": Object {
                  "result": 2,
                  "weight": 107,
                },
                "squat": Object {
                  "result": 0,
                  "weight": 174,
                },
              },
            ]
          }
        />
      </div>
      <MacroTracker
        macros={
          Object {
            "current": Object {
              "carbs": 130,
              "fat": 80,
              "protein": 157,
            },
            "target": Object {
              "carbs": 280,
              "fat": 80,
              "protein": 340,
            },
          }
        }
        setMacros={[Function]}
        updateMacro={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`should render Home Page with welcome message 1`] = `
<Fragment>
  <h2>
    Welcome to flext!
  </h2>
  <p>
    Let's begin by entering your starting weights
  </p>
  <Link
    to="/onboarding"
  >
    <Button
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Enter Weights
    </Button>
  </Link>
</Fragment>
`;

exports[`should render Home Page without props 1`] = `
<div
  className="page--with-side-nav"
>
  <SideNav
    path=""
  />
  <div
    className="side-nav__page-content"
  >
    <div
      className="side-nav__page-scrim side-nav__page-scrim--hidden"
    />
    <div
      className="dashboard__container side-nav__page-content"
    >
      <NextWorkout
        liftVariant={
          Object {
            "a": 0,
            "b": 0,
          }
        }
        masterWeights={
          Object {
            "bench": 0,
            "chinup": Object {},
            "deadlift": 0,
            "overhead": 0,
            "row": 0,
            "squat": 0,
          }
        }
        onStartWorkout={[Function]}
      />
      <div
        className="card"
      >
        <Calendar
          onClickWorkoutDate={[Function]}
          workouts={Array []}
        />
      </div>
      <MacroTracker
        macros={
          Object {
            "current": Object {
              "carbs": 0,
              "fat": 0,
              "protein": 0,
            },
            "target": Object {
              "carbs": 0,
              "fat": 0,
              "protein": 0,
            },
          }
        }
        setMacros={[Function]}
        updateMacro={[Function]}
      />
    </div>
  </div>
</div>
`;
